src/writer/api/clients.py

```
@@ -0,0 +1,158 @@
+        Internal method to set a key-value pair in the agent data.
+        :param key: The key to set.
+        :param value: The value to set for the key.
+        :param user_id: Optional user ID for the request.
+        :return: The response from the API or None if failed.
+        """
+        data = {"key": key, "data": value}
+
+        url = "agent_data"
+        response = self.request("POST", url, data, user_id)
+        if response is None:
+            logging.error(f"Failed to set value for key: {key}")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)